### PR TITLE
feat(anta): Add enable_password in inventory and convert TestResult.message to a list of str

### PR DESCRIPTION
**Pull request by titom73**

### CODE REVIEW

File: anta/inventory/__init__.py [hunk 1]
```diff
@@ -93,8 +94,7 @@ def __init__(self, inventory_file: str, username: str, password: str, auto_conne
             auto_connect (bool, optional): Automatically build eAPI context for every devices. Defaults to True.
             timeout (float, optional): Timeout in second to wait before marking device down. Defaults to 5sec.
         """
-        self._username = username
-        self._password = password
+        self.set_credentials(username,password,enable_password)
```
Comment (gmuloc) [inline anta/inventory/__init__.py:97]:
```suggestion
        self.set_credentials(username, password, enable_password)
```

File: anta/result_manager/report.py [hunk 2]
```diff
@@ -46,6 +46,12 @@ def __init__(self, headers=None, colors: bool = True) -> None:
         self.colors = colors
         self.headers = headers if headers is not None else self.DEFAULT_HEADERS
 
+    def _split_list_to_txt_list(self, usr_list: List[str], delimiter: str = '*'):
+        result = ''
+        for line in usr_list:
+            result = f'{result}\n{delimiter} {line}' if result != '' else f'{delimiter} {line}'
+        return result
```
Comment (gmuloc) [inline anta/result_manager/report.py:53]:
```suggestion
    def _split_list_to_txt_list(self, usr_list: List[str], delimiter: str = '*'):
        return "\n".join(f"{delimiter} {line}" for line in usr_list)
```

**Review by ksator — DISMISSED** [review]
we are importing twice multiprocessing
```
import multiprocessing
from multiprocessing import Pool
multiprocessing.cpu_count
with Pool
```
is it not better to do that?
```
import multiprocessing
multiprocessing.cpu_count
with multiprocessing.Pool

**Comment by ksator** [comment]
tested using
```
from anta.inventory import AntaInventory
import anta.tests
inventory = AntaInventory(inventory_file="examples/inventory.yml", username="arista", password="arista99t7", auto_connect=True, timeout=2, enable_password="cw")
devices = inventory.get_inventory(established_only = True)
anta.tests.verify_ntp(devices[0].session,devices[0].enable_password)
anta.tests.verify_running_config_diffs(devices[0].session,devices[0].enable_password)
```

**Review by ksator — APPROVED** [review]
LGTM

**Pull request merged**